pennylane/numpy/tensor.py

```diff
@@ -113,6 +113,43 @@ def __array_wrap__(self, obj):
         out_arr = tensor(obj, requires_grad=self.requires_grad)
         return super().__array_wrap__(out_arr)
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # pylint: disable=no-member,attribute-defined-outside-init
+        outputs = [
+            i.view(onp.ndarray) if hasattr(i, "unwrap") else i for i in kwargs.get("out", ())
+        ]
+
+        if outputs:
+            outputs = tuple(outputs)
+            kwargs["out"] = outputs
+        else:
+            outputs = (None,) * ufunc.nout
+
+        # unwrap the input arguments to the ufunc
+        args = [i.unwrap() if hasattr(i, "unwrap") else i for i in inputs]
+
+        # call the ndarray.__array_ufunc__ method to compute the result
+        # of the vectorized ufunc
+        res = super().__array_ufunc__(ufunc, method, *args, **kwargs)
+
+        if ufunc.nout == 1:
+            res = (res,)
+
+        res = [
+            (onp.asarray(result) if output is None else output)
+            for result, output in zip(res, outputs)
+        ]
+
+        # if any of the inputs were trainable, the output is also trainable
+        requires_grad = any(
+            isinstance(x, onp.ndarray) and getattr(x, "requires_grad", True) for x in inputs
+        )
+
+        for i in range(len(res)):  # pylint: disable=consider-using-enumerate
+            res[i] = tensor(res[i], requires_grad=requires_grad)
+
+        return res[0] if len(res) == 1 else tuple(res)

```

Comment:
Arg I am really not clued up on this module...why would you return a tuple here? Not the onp.array?

Reply:
This is a restriction on NumPy ufuncs - they either return a single numpy array, or a tuple of arrays. This echoes how Python functions work.

Reply:
I'll add comments

Reply:
Have added some comments, let me know if it makes it easier to follow!

Reply:
Yes, thanks!